server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -202,8 +202,17 @@ public class SnapshotsService extends AbstractLifecycleComponent implements Clus
         Setting.Property.Dynamic
     );
 
+    public static final Setting<Boolean> SNAPSHOT_V2 = Setting.boolSetting(
```

Comment:
can we move to repo in this version itself ?

Reply:
I think we should go with cluster setting rather than repository setting. It will help us to make a decision for the code path to execute at https://github.com/opensearch-project/OpenSearch/blob/a6090a7b8b482aa853af1d03f28ea0d6b4025e03/server/src/main/java/org/opensearch/action/admin/cluster/snapshots/create/TransportCreateSnapshotAction.java#L107. Any specific reason we want to go with repo setting?

Reply:
Ack